exploration/contextual-options.md

```diff
@@ -0,0 +1,156 @@
+		<dd><a href="https://github.com/unicode-org/message-format-wg/pull/780">#780</a></dd>
+	</dl>
+</details>
+
+## Objective
+
+Provide common definitions for contextual options.
+
+## Background
+
+Function options may influence the resolution, selection, and formatting of annotated expressions.
+These provide a great solution for options like `minFractionDigits`, `dateStyle`,
+or other similar factors that influence the formatted result.
+
+Such options naturally correspond to function arguments or builder-style function constructors.
+Each option is specific to the associated API.
+Users (message authors, translators, developers) should not expect
+that a given option name has the same meaning between functions
+or that its behavior stays the same.
+
+To reduce the learning curve for users and improve consistency,
+it would be useful to have common options 
+(generally those related to the formatting context)
+shared between all functions.
+
+## Use-Cases
+
+At least the following function and markup options should be considered:
+
+- An identifier for the expression or markup.
+  This is included in the formatted part,
+  and allows the parts of an expression to be explicitly addressed.
+
+  > Example identifying two literal numbers:
+  >
+  > ```
+  > The first number was {1234 :number u:id=first} and the second {56789 :number u:id=second}.
+  > ```
+
+- An override for the locale used to format the expression.
+
+  > Example embedding a French date in an English message:
+  >
+  > ```
+  > In French, this date would be displayed as {|2024-05-06| :date u:locale=fr}
+  > ```
+
+- An override for the LTR/RTL/auto directionality of the expression.
+
+  > Example explicitly isolating the directionality of a placeholder
+  > for a custom user-defined function:
+  >
+  > ```
+  > Welcome, {$user :x:username u:dir=auto}
+  > ```
+
+## Requirements
+
+Common options or attributes should work the same way in different functions.
+
+Special options or attributes should not conflict with other option names.
+
+## Constraints
+
+User-defined functions are free to use un-namespaced option names
+even if the function identifier does use a namespace.
+
+The `u` namespace is reserved for future standardization.
+
+No literal syntax is provided for values that are lists or sequences of scalar values.
+
+## Proposed Design
+
+Define the expected values and handling for the following options
+wherever they are used:
+
+- `u:id` — A string value that is included as an `id` or other suitable value
+  in the formatted part(s) for the placeholder,
+  or any other structured formatted results.
+  Ignored when formatting to a string, but could show up in error messages.
+- `u:locale` — A comma-delimited list of BCP 47 language tags,
+  or an implementation-defined list of such tags.
+  The tags are parsed, and they replace the _locale_
+  defined in the _formatting context_ for this expression or markup.
+- `u:dir` — One of the string values `ltr`, `rtl`, or `auto`.
```

Comment:
Might not be enough?
I think that some kind if "isolate" would be handy.

The equivalent of `<bdi>` in HTML. Not sure whey that is a separate concept in HTML, and I suspect history / legacy.

The `auto` in HTML:
> `auto`, which lets the user agent decide. It uses a basic algorithm as it parses the characters inside the element until it finds a character with a strong directionality, then applies that directionality to the whole element.
> https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir

That sounds an awful lot like `bdi`, but less clear:
> Embedding the characters in `<span dir="auto">` has the same effect as using `<bdi>`, but its semantics are less clear.
> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/bdi